src/field/pokemon.ts

```diff
@@ -5361,30 +5363,30 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
               defType,
             );
             if (cancelImmunity.value) {
-              return false;
+              return this.messageIsImmune(quiet);
```

Comment:
```suggestion
              return false;
```
You're in a `.map` call here, not the greater function. `cancelImmunity` is used to check if the user has corrosion, so the result here would be playing "It doesn't effect {pokemon}" when corrosion successfully lands a poison on a poison/steel type. This we clearly don't want

Reply:
When i was making the change i actually didn't notice it was inside the mapping function and the tests i ran also didn't show any error. So it was actually my mistake, thank you for pointing it out.